src/libstrongswan/plugins/pkcs11/pkcs11_library.c

```diff
@@ -18,7 +18,13 @@
 
 #include "pkcs11_library.h"
 
-#include <dlfcn.h>
+#ifdef WIN32
+#  include <utils/compat/windows.h>
+/* windows.h defines it as CreateMutexA, which breaks compilation */
+#  undef CreateMutex
```

Comment:
I'm not sure if undefining `CreateMutex` is really needed here, when no `<windows.h>` is included, but it won't harm anyway.

Reply:
It is, because the header is included by our compat layer.

Reply:
Yes, not directly, but `<windows.h>` is included from `<winsock2.h>`.

Reply:
And I would keep inclusion of `<utils/compat/windows.h>` in `gh193-pkcs11-windows` for clarity.

Reply:
No, we never include such headers directly only `utils.h`, which is included in `pkcs11_library.h` already. Also, the `#undef` is only needed IF it is included, so why would we explicitly do so?